src/idfxt/JSONDataInterface.h

```diff
@@ -0,0 +1,102 @@
+// copyright 2014, Shannon Mackey <mackey@BUILDlab.net>
+#ifndef JSONDATAINTERFACE_H
+#define JSONDATAINTERFACE_H
+
+//for JSONDataObject and JSONDataInterface
+#include <string>
+#include <list>
+#include <map>
+#include <memory>
+
+//for old IDF text import only
+#include <algorithm>
+#include <cctype>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <chrono>
+#include <random>
+
+const std::string ACCEPTED_VERSION = "8.2";
+//end old IDF text import only
+
+extern "C"
+{
+#include "../../third_party/cJSON/cJSON.h"
+}
+
+namespace idfx {
+
+
+class JSONDataObject
+{
+public:
+    JSONDataObject(const std::string &json_content);
+    ~JSONDataObject();
+
+    std::map<std::string, std::string> getProperties();
+    std::string getPropertyValue(std::string);
+//    void setPropertyValue(std::string, std::string);
+//    void setPropertyValue(std::string, int);
+//    void setPropertyValue(std::string, double);
+
+    std::list<std::shared_ptr<JSONDataObject> > getExtensions();
```

Comment:
I initially had Children.  But, that suggested generic nested object, and I
really meant to convey, by using the current parlance  that these,
specifically, are the extensible bits of the old IDF.

On Fri, Jan 16, 2015 at 6:43 PM, Dan Macumber <notifications@github.com>
wrote:

> In src/idfxt/JSONDataInterface.h
> <https://github.com/NREL/EnergyPlus/pull/4636#discussion-diff-23117683>:
>
> > +namespace idfx {
> > +
> > +
> > +class JSONDataObject
> > +{
> > +public:
> > +    JSONDataObject(const std::string &json_content);
> > +    ~JSONDataObject();
> > +
> > +    std::map<std::string, std::string> getProperties();
> > +    std::string getPropertyValue(std::string);
> > +//    void setPropertyValue(std::string, std::string);
> > +//    void setPropertyValue(std::string, int);
> > +//    void setPropertyValue(std::string, double);
> > +
> > +    std::list<std::shared_ptr<JSONDataObject> > getExtensions();
>
> @refaqtor <https://github.com/refaqtor> is there a better word that
> extensions? children?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NREL/EnergyPlus/pull/4636/files#r23117683>.
>